### ISSUE TITLE
Write a dbplyr version of clean_names?

### ISSUE THREAD

**Issue by sfirke** [sfirke/janitor#467]
I love how with the `dbplyr` package I can use dplyr verbs on SQL Server tables like this:
```
compost_inbound <- tbl(uat, "SolidWasteParadigmOnly") %>%
  filter(RouteMainDesc == 'AA RES COMPOST', MT %in% c(8001, 8030)) %>%
  collect()
```
But I wish I could add ` %>% clean_names` at the start of that pipeline!

I'm using Microsoft SQL Server every day but am a beginner at programming it.

I have no idea how hard this would be to do, how much it would bloat janitor's dependencies, and if there's another package that it would belong in.  But I wish it existed.  Does anyone know about programming `dbplyr` functions?

**Comment by billdenney** [comment]
While I do a lot in the `clean_names()` space, I don't do almost any database work.

**Comment by sfirke** [comment]
I posted this over in the dbplyr issue tracker: https://github.com/tidyverse/dbplyr/issues/752  Folks there will hopefully be able to tell me how hard this would be and if anyone who knows how to do it wants to help.

**Comment by sfirke** [comment]
Okay this might be very easy, see: https://github.com/tidyverse/dbplyr/issues/752#issuecomment-1012999134

**Comment by sfirke** [comment]
Leaving this issue open so I don't lose sight of it but I think I should probably add it to a new package at a new repo, maybe `dbjanitor`.  That package would never go to CRAN but would be installable from GitHub.  Then I don't have to add `dbplyr` as a dependency to janitor and worry about keeping 100% unit test coverage here, etc.

If/when I do that, or decide I never will, I'll close this.

**Comment by sfirke** [comment]
~~Given how simple the function is in the comment linked above, I think folks who want this functionality can just use that code and define the function in their scripts.  It's not enough to justify a new package or adding a dependency to janitor.~~

Wait, one last thought.  janitor::clean_names works on `sf` and `tidygraph` objects but doesn't depend on them, they're in `Suggests`.  Could I do this the same way with dbplyr to expand the functionality of `clean_names` without requiring another dependency?

**Comment by billdenney** [comment]
It should work.

**Comment by sfirke** [comment]
I will see how it's done for `sf` and replicate with a PR.  Others wrote the `sf` and `tidygraph` implementations so this'll be new for me.

**Comment by billdenney** [comment]
I think that the general idea is to add `dbplyr` to the DESCRIPTION file under `Suggests`, make the test use `skip_if_not_installed("dbplyr")`, and add an `if (!requireNamespace("dbplyr") {stop("dbplyr must be installed")` to the function itself.

**Comment by sfirke** [comment]
implemented in #507
